pysteps/tests/test_nowcasts_lagrangian_probability.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+
+from pysteps.nowcasts.lagrangian_probability import forecast
+from pysteps.tests.helpers import get_precipitation_fields
+from pysteps.motion.lucaskanade import dense_lucaskanade
+
+
+def test_numerical_example():
+    """"""
+    precip = np.zeros((20, 20))
+    precip[5:10, 5:10] = 1
+    velocity = np.zeros((2, *precip.shape))
+    timesteps = 4
+    thr = 0.5
+    slope = 1  # pixels / timestep
+
+    # compute probability forecast
+    fct = forecast(precip, velocity, timesteps, thr, slope=slope)
+
+    assert fct.ndim == 3
+    assert fct.shape[0] == timesteps
+    assert fct.shape[1:] == precip.shape
+    assert fct.max() <= 1.0
+    assert fct.min() >= 0.0
+
+
+def test_real_case():
+    """"""
+    # inputs
+    precip, metadata = get_precipitation_fields(
+        num_prev_files=2,
+        num_next_files=0,
+        return_raw=False,
+        metadata=True,
+        upscale=2000,
+    )
+
+    # motion
+    motion = dense_lucaskanade(precip)
+
+    # parameters
+    timesteps = [1, 2, 3]
+    thr = 1  # mm / h
+    slope = 1 * metadata["accutime"]  # min-1
+
+    # compute probability forecast
+    extrap_kwargs = dict(allow_nonfinite_values=True)
```

Comment:
What is this option for?

Reply:
it allows extrapolating a field with missing values

Reply:
it allows extrapolating a field with missing values